src/sql/src/plan/typeconv.rs

```diff
@@ -175,9 +175,15 @@ lazy_static! {
 
             // REGPROC
             (RegProc,Oid) => Implicit: CastRegProcToOid,
+            (RegProc, String) => Explicit: sql_impl_cast("(
+                SELECT DISTINCT(name) FROM mz_catalog.mz_functions WHERE oid = $1
+            )"), // DISTINCT is required to handle overridden functions such as generate_series
 
             // REGTYPE
             (RegType,Oid) => Implicit: CastRegTypeToOid,
+            (RegType, String) => Explicit: sql_impl_cast("(
+                SELECT name FROM mz_catalog.mz_types WHERE oid = $1
```

Comment:
This doesn't handle the case where `$1` doesn't exist. Here's what I came up with:

```
            (RegType, String) => Explicit: sql_impl_cast("(
                SELECT COALESCE(t.name, v.x::pg_catalog.text)
                FROM
                    (VALUES ($1::pg_catalog.oid)) AS v(x)
                    LEFT JOIN mz_catalog.mz_types AS t ON t.oid = v.x
            )"),
```

Probably ditto for regproc.

Reply:
this is what you get in PG
```
martin=# select 1::regtype::text;
 text 
------
 1
(1 row)
```

This would be ideal:
```
SELECT ISNULL(name, $1::text) FROM mz_catalog.mz_types WHERE oid = $1;
ERROR:  function isnull(text, text) does not exist
```

Reply:
The query I posted above works.